webapp/app/js/perf.js

```diff
@@ -203,35 +217,57 @@ perf.factory('PhCompare', [ '$q', '$http', 'thServiceDomain', 'PhSeries', 'math'
       return cmap;
     },
 
-    getResultsMap: function(projectName, seriesList, timeRange, resultSetId) {
+    getInterval: function(oldTimestamp, newTimestamp) {
+      var now = (new Date()).getTime() / 1000;
+      var timeRange = Math.min(oldTimestamp, newTimestamp);
+      timeRange = Math.round(now - timeRange);
+
+      //now figure out which predefined set of data we can query from
+      var intervals = [86400, 604800, 1209600, 2592000, 5184000, 7776000];
+      return _.find(intervals, function(i) { return timeRange <= i });
+    },
+
+    getResultsMap: function(projectName, seriesList, timeRange, resultSetIds) {
       var baseURL = thServiceDomain + '/api/project/' +
         projectName + '/performance-data/0/' +
         'get_performance_data/?interval_seconds=' + timeRange;
 
       var resultsMap = {};
-      return $q.all(seriesList.map(function(series) {
-        return $http.get(baseURL + "&signatures=" + series.signature).then(
+      var chunks = _.chunk(seriesList, 20);
+      return $q.all(chunks.map(function(seriesChunk) {
```

Comment:
May as well make this all one expression: return $q.all(_chunk(seriesList, 20).map(...)